### PR TITLE
Re-use existing selectors in ScanQuery and SelectQuery

### PR DESCRIPTION
**Pull request by joykent99**
ScanQuery and SelectQuery already construct a list of selectors. This changelist
re-uses those selectors instead of creating new ones.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/scan/ScanQueryEngine.java [hunk 1]
```diff
@@ -223,7 +213,7 @@ public void remove()
                               for (int j = 0; j < allColumns.size(); j++) {
                                 theEvent.add(getColumnValue(j));
                               }
-                              UsageUtils.incrementAuSignals((AtomicLong) responseContext.get(UsageUtils.NUM_AU_SIGNALS), columnValueSelectors);
+                              UsageUtils.incrementAuSignals((AtomicLong) responseContext.get(UsageUtils.NUM_AU_SIGNALS), columnSelectors);
```
Comment (liliuleo93) [inline processing/src/main/java/org/apache/druid/query/scan/ScanQueryEngine.java]: Good to reuse, but there is a problem, for example, a scan query can be `select <columns>, <virtual columns> from xx where <filter>`, and `columnSelectors` only contains `columns` and `virtual column`, `filter` is left. And `virtual column` is not `required columns of virtual column`, so duplication can happened

File: processing/src/main/java/org/apache/druid/query/select/SelectQueryEngine.java [hunk 2]
```diff
@@ -260,12 +253,17 @@ public void addRowValuesToSelectResult(
               builder.addDimension(dimSpec.getOutputName());
             }
 
+            for (ColumnSelectorPlus<SelectColumnSelectorStrategy> selectorPlus : selectorPlusList) {
+              columnValueSelectors.add(selectorPlus.getSelector());
```
Comment (liliuleo93) [inline processing/src/main/java/org/apache/druid/query/select/SelectQueryEngine.java]: The same as `ScanQuery`.

File: processing/src/main/java/org/apache/druid/query/scan/ScanQueryEngine.java (hunk 1 above)
Comment (joykent99) [reply processing/src/main/java/org/apache/druid/query/scan/ScanQueryEngine.java]: Good point, @liliuleo93. Let us keep what you have for now then. We might re-visit this when we find out more.

**Comment by joykent99** [comment]
Close this PR for now.

**Pull request closed without merging**